vm/src/stdlib/os.rs

```diff
@@ -451,13 +499,25 @@ fn os_stat(path: PyStringRef, vm: &VirtualMachine) -> PyResult {
 }
 
 #[cfg(unix)]
-fn os_symlink(src: PyStringRef, dst: PyStringRef, vm: &VirtualMachine) -> PyResult<()> {
+fn os_symlink(
+    src: PyStringRef,
+    dst: PyStringRef,
+    dir_fd: DirFd,
+    vm: &VirtualMachine,
+) -> PyResult<()> {
     use std::os::unix::fs as unix_fs;
+    let src = make_path(vm, src, &dir_fd);
```

Comment:
The `dir_fd` in symlink is only relevant for the `dst`. The underlying implementation use [`symlinkat`](https://linux.die.net/man/2/symlinkat).